packages/interactivity/src/directives.js

```diff
@@ -278,4 +285,55 @@ export default () => {
 			} );
 		}
 	);
+
+	// data-wp-slot
+	directive(
+		'slot',
+		( {
+			directives: {
+				slot: { default: slot, above, below },
+			},
+			props: { children },
+		} ) => {
+			return (
+				<Fragment>
+					{ above && <Slot name={ above } /> }
+					{ slot ? (
+						<Slot name={ slot }>{ children }</Slot>
+					) : (
+						children
+					) }
+					{ below && <Slot name={ below } /> }
+				</Fragment>
+			);
+		},
+		{ priority: 4 }
+	);
+
+	// data-wp-fill
+	directive(
+		'fill',
+		( {
+			directives: {
+				fill: { default: fill },
+			},
+			props: { children },
+			evaluate,
+			context,
+		} ) => {
+			const contextValue = useContext( context );
+			const slot = evaluate( fill, { context: contextValue } );
+			return <Fill slot={ slot }>{ children }</Fill>;
+		},
+		{ priority: 4 }
+	);
+
+	// data-wp-slot-provider
+	directive(
+		'slot-provider',
+		( { props: { children } } ) => (
+			<SlotProvider>{ children }</SlotProvider>
+		),
+		{ priority: 4 }
```

Comment:
@darerodz: Is there a use case where `data-wp-slot-provider` and `data-wp-slot` can be used in the same component, or that doesn't make sense?